### PR TITLE
word2vec new version

### PR DESCRIPTION
**Pull request by frankwhzhang**

### CODE REVIEW

File: fluid/PaddleRec/word2vec/net.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+    datas.append(input_word)
+    datas.append(true_word)
+    datas.append(neg_word)
+
+    py_reader = fluid.layers.create_py_reader_by_data(
+        capacity=64, feed_list=datas, name='py_reader', use_double_buffer=True)
+
+    words = fluid.layers.read_file(py_reader)
+    init_width = 0.5 / embedding_size
```
Comment (gavin1332) [inline fluid/PaddleRec/word2vec/net.py]: Is "init_range" much better? It is not so intuitive to follow until I find fluid.initializer.Uniform(-init_width, init_width)

File: fluid/PaddleRec/word2vec/README.md [hunk 2]
```diff
@@ -1,103 +1,73 @@
...
-a
-b
-c
-```
-Then, please set --other_dict_path as the directory of where you
-save the vocab you will use and set --with_other_dict flag on to using it.
 
-## Train
-The command line options for training can be listed by `python train.py -h`.
+根据词典将文本转成id, 同时进行downsample，按照概率过滤常见词。
```
Comment (guru4elephant) [inline fluid/PaddleRec/word2vec/README.md]: 这个downsample是离线做一次即可？

File: fluid/PaddleRec/word2vec/net.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    neg_word_reshape = fluid.layers.reshape(words[2], shape=[-1, 1])
+    neg_word_reshape.stop_gradient = True
+
+    neg_emb_w = fluid.layers.embedding(
+        input=neg_word_reshape,
+        is_sparse=is_sparse,
+        size=[dict_size, embedding_size],
+        param_attr=fluid.ParamAttr(
+            name='emb_w', learning_rate=10.0))
+    # param_attr='emb_w')
```
Comment (guru4elephant) [inline fluid/PaddleRec/word2vec/net.py]: remove this comment please.

File: fluid/PaddleRec/word2vec/net.py [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+    neg_emb_w_re = fluid.layers.reshape(
+        neg_emb_w, shape=[-1, neg_num, embedding_size])
+    neg_emb_b = fluid.layers.embedding(
+        input=neg_word_reshape,
+        is_sparse=is_sparse,
+        size=[dict_size, 1],
+        param_attr=fluid.ParamAttr(
+            name='emb_b', learning_rate=10.0))
+    #param_attr='emb_b')
```
Comment (guru4elephant) [inline fluid/PaddleRec/word2vec/net.py]: remove this comment.

File: fluid/PaddleRec/word2vec/net.py [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+            fluid.layers.elementwise_mul(input_emb, true_emb_w),
+            dim=1,
+            keep_dim=True),
+        true_emb_b)
+    input_emb_re = fluid.layers.reshape(
+        input_emb, shape=[-1, 1, embedding_size])
+    neg_matmul = fluid.layers.matmul(
+        input_emb_re, neg_emb_w_re, transpose_y=True)
+    neg_matmul_re = fluid.layers.reshape(neg_matmul, shape=[-1, neg_num])
+    #neg_matmul_reshape = fluid.layers.reshape(neg_matmul, shape=[-1, ]
```
Comment (hutuxian) [inline fluid/PaddleRec/word2vec/net.py]: remove this comment?

File: fluid/PaddleRec/word2vec/train.py [hunk 6]
```diff
@@ -261,123 +193,42 @@ def train(args):
...
 
     logger.info("dict_size: {}".format(word2vec_reader.dict_size))
+    #update id_frequency
+    sum_val = 0.0
+    id_frequencys_pow = []
+    for id_x in range(len(word2vec_reader.id_frequencys)):
+        id_frequencys_pow.append(
+            math.pow(word2vec_reader.id_frequencys[id_x], 0.75))
+        sum_val += id_frequencys_pow[id_x]
+    id_frequencys_pow = [val / sum_val for val in id_frequencys_pow]
```
Comment (hutuxian) [inline fluid/PaddleRec/word2vec/train.py]: Maybe it's more clear to use numpy?
```suggestion
    id_frequencys_pow = np.power(word2vec_reader.id_frequencys, 0.75)
    id_frequencys_pow /= np.sum(id_frequencys_pow)
```

File: fluid/PaddleRec/word2vec/net.py [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+        true_logits, shape=[-1, neg_num], value=0.0, dtype='float32')
+
+    true_xent = fluid.layers.sigmoid_cross_entropy_with_logits(true_logits,
+                                                               label_ones)
+    neg_xent = fluid.layers.sigmoid_cross_entropy_with_logits(neg_logits,
+                                                              label_zeros)
+
+    cost = fluid.layers.elementwise_add(
+        fluid.layers.reduce_sum(
+            true_xent, dim=1),
```
Comment (hutuxian) [inline fluid/PaddleRec/word2vec/net.py]: Can we remove this `reduce_sum` since the second dimension of `true_xent` is always one?

File: fluid/PaddleRec/word2vec/README.md (hunk 2 above)
Comment (frankwhzhang) [reply fluid/PaddleRec/word2vec/README.md]: 离线一次就行

**Review by gavin1332 — APPROVED** [review]
